src/IntegrationUtils.ts

```diff
@@ -59,13 +60,53 @@ const choiceList = [
 
  let childProcessMap : Map<string, child_process.ChildProcess>;
 
- export function startIntegration(context: vscode.Uri): Promise<boolean> {
+ export function startIntegration(...args: any[]): Promise<boolean> {
 	return new Promise <boolean> ( async (resolve, reject) => {
-		const choice : string | undefined = await vscode.window.showQuickPick(choiceList, {
-			placeHolder: 'Select the type of Apache Camel K Integration'
-		});
+		let context : vscode.Uri | undefined;
+		let inChoice : string | undefined;
+
+		if (args && args.length > 0) {
+			// if there are multiple arguments, assume didact input
+			// at this time we only support the URI & basic/dev mode type of integration
+			if (args[0] instanceof vscode.Uri) {
+				context = args[0];
+				inChoice = undefined;
+			} else if (Array.isArray(args[0])) {
+				let innerArgs1 : any[] = args[0];
+				let innerArgs2 : any[] = innerArgs1[0];
```

Comment:
this is considerably complexifying the code. I spent 20 minutes to understand what it is doing.

I don't have an easy solution.
The only little thing that i found that might help is to provide more precise signature, something like: ` export function startIntegration(args: [vscode.Uri | [[vscode.Uri, string?]]]): Promise<boolean> {`

Reply:
I cannot change the signature. As I pointed out earlier, regardless of what I put in the function argument declaration, the first argument always comes through as an array of arguments. It appears that execute isn't distinguishing between a context-only argument list and a mix of arguments... it's always passed an array of any[]...

executeCommand<T>(command: string, ...rest: any[]): Thenable<T | undefined>

I have tried multiple ways to make it less complex. It cannot change. The nested series of arrays is crazy, I know.

Reply:
I tried with `export function startIntegration(args: [vscode.Uri | [[vscode.Uri, string?]]]): Promise<boolean> {` and it seems to work

Reply:
that seems to work for what exactly? that's not any less confusing to me

Reply:
it allows to have a signature which is more precise than an array of anything. it is shows that it is supporting an array of vscode.Uri or an array of array of array with first a vscode.Uri and then a potential string. But it is not even true. i don't understand why it was working when i used it.
I'm really confused by this code and the TypeScript notation.

Reply:
No, I understand what you were after by changing the signature. I don't disagree that it's confusing. I'm just saying that I tried various combinations and none of them worked consistently. I think that it's a case where you have to delete the out folder and recompile because you're changing the signature -- and that in turn changes the TypeScript to JavaScript conversion...